### PR TITLE
GitHub Actions; Refactor tests

### LINKED ISSUE #525: Move tests from travis to github actions
Because reasons.

### LINKED ISSUE #526: Run rtd upon pr via pr webhook

### PR DESCRIPTION
**Pull request by yannikschaelte**
fixes #525 (github actions), fixes #526 (rtd tests)

### CODE REVIEW

**Review by FFroehlich — APPROVED** [review]

**Comment by yannikschaelte** [comment]
No idea why codecov says coverage decreased. On their page it seems to be more, i.e. all (base, petab, optimize) should have been uploaded and correctly merged.

File: .github/workflows/install_deps.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
+
```
Comment (dweindl) [inline .github/workflows/install_deps.sh:2]: 1-line summary?

File: .github/workflows/ci_pr.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
+name: CI
+
+# trigger
+on: [pull_request]
+
+jobs:
+  notebooks1:
```
Comment (dweindl) [inline .github/workflows/ci_pr.yml:7]: Maybe add some comment about the scope of each job and where new things should (not) be added?

File: test/petab/test_petab_import_pysb.py [hunk 3]
```diff
@@ -3,22 +3,27 @@
...
 # must import after previous, otherwise circular import issues :(
 from amici.petab_import_pysb import PysbPetabProblem
+import petabtests
+
+# In CI, bionetgen is install here
+BNGPATH = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), '..', '..', 'BioNetGen-2.3.2'))
+if 'BNGPATH' not in os.environ:
+    logging.warning(f"Env var BNGPATH was not set. Setting to {BNGPATH}")
+    os.environ['BNGPATH'] = BNGPATH
```
Comment (dweindl) [inline test/petab/test_petab_import_pysb.py:21]: Not optimal for people who might have BNG on `PATH` but no `BNGPATH` set. But not sure how many people that will affect... So probably okay.

File: test/run_notebook.sh [hunk 4]
```diff
@@ -1,10 +1,38 @@
 #!/bin/bash
-# Run jupyter notebooks as given on command line, show output only on error. 
-# If a directory is provided, run all contained notebooks non-recursively.
+
+# Run notebooks
```
Comment (dweindl) [inline test/run_notebook.sh:3]: Document CLI options?

**Review by dweindl — APPROVED** [review]
Great! Looks good from what I can tell.

File: test/petab/test_petab_import_pysb.py (hunk 3 above)
Comment (yannikschaelte) [reply test/petab/test_petab_import_pysb.py:21]: true ... well, easy to change if it's a problem, leaving for the moment.

File: test/base/test_objective.py [hunk 5]
```diff
@@ -231,15 +143,15 @@ def test_finite_difference_checks():
 
     objective = pypesto.Objective(fun=fun, grad=grad)
 
-    def rel_err(eps):
+    def rel_err(eps_):
```
Comment (jvanhoefer) [inline test/base/test_objective.py:146]: why did you rename the variable to `eps_`? Looks unexpected to me :D

**Review by jvanhoefer — APPROVED** [review]
Really cool!

File: test/base/test_objective.py (hunk 5 above)
Comment (yannikschaelte) [reply test/base/test_objective.py:146]: just to avoid shadowing
